File: solutions/csharp/log-levels/2/LogLevels.cs
static class LogLine
{
    public static string Message(string logLine) => logLine.Remove(0, logLine.IndexOf(":") + 1).Trim();


    public static string LogLevel(string logLine)
    {
        var strings = logLine.Split();
        return strings[0].Trim([':', '[', ']']).ToLower();
    }

    // public static string Reformat(string logLine) => Message(logLine) + " (" + LogLevel(logLine) + ")";
    public static string Reformat(string logLine) => $"{Message(logLine)} ({LogLevel(logLine)})";
}

File: solutions/java/log-levels/1/src/main/java/LogLevels.java
public class LogLevels {
    
    public static String message(String logLine) {
        int logLevelIndex = logLine.indexOf("]");
        return logLine.substring(logLevelIndex + 2).trim();
    }

    public static String logLevel(String logLine) {
        int logLevelIndex = logLine.indexOf("]");
        return logLine.substring(1, logLevelIndex).toLowerCase();
    }

    public static String reformat(String logLine) {
        return message(logLine) + " (" + logLevel(logLine) + ")";
    }
}
